libs/deque/src/main.sw

```diff
@@ -0,0 +1,75 @@
+library;
+//! The `Deque` type corresponds to the same called data structure.
+//! A Deque is defined as a linear data structure that allows insertion and removal of elements from both ends.
+
+pub struct Deque<T> {
+    front: Vec<T>,
+    back: Vec<T>,
+}
+
+impl<T> Deque<T> {
+    /// Creates a new, empty deque.
+    pub fn new() -> Self {
+        Self {
+            front: Vec::new(),
+            back: Vec::new(),
+        }
+    }
+
+    /// Creates a new, empty deque with the specified capacity.
+    pub fn with_capacity(capacity: u64) -> Self {
+        Self {
+            front: Vec::with_capacity(capacity),
+            back: Vec::with_capacity(capacity),
+        }
+    }
+
+    /// Appends an element to the back of the deque.
+    pub fn push_back(ref mut self, value: T) {
+        self.back.push(value);
+    }
+
+    /// Inserts an element at the front of the deque.
+    pub fn push_front(ref mut self, value: T) {
+        self.front.push(value);
+    }
+
+    pub fn pop_front(ref mut self) -> Option<T> {
+        if self.front.is_empty() {
+            if !self.back.is_empty() {
+                self.front.push(self.back.remove(0))
```

Comment:
```suggestion
               self.back.remove(0)
```

Is there a reason that we do not actually remove from the `front`? Here we remove the first item from the `back` but then we add it to the end of the `front` which effectively does not pop the item.

May need to run `forc fmt` here. and same below